### PR TITLE
MDEV-15442 xtrabackup-v2 SST donor stuck in DONOR/DESYNCED state when…

### PR DESCRIPTION
**Pull request by mkaruza**
Monitor mysqld process and if it is terminated during sst stream kill xtrabackup-v2 script and children

### CODE REVIEW

**Comment by svoj** [comment]
Hi @mkaruza,

Thanks for your contribution. Corresponding JIRA task: https://jira.mariadb.org/browse/MDEV-15442

This task was added to 10.1  backlog.

Similar to other open source projects, the MariaDB Foundation needs to have shared ownership of all code that is included in the MariaDB distribution. The easiest way to achieve this is by submitting your code under the BSD-new license.

The other alternative is to sign the code contribution agreement which can be found here: https://mariadb.com/kb/en/mariadb/mca/

Please indicate in a comment below that you are contributing your new code of the whole pull request, including one or several files that are either new files or modified ones, under the BSD-new license or that you have filled out the contribution agreement and sent it.

Thanks,
Sergey

**Comment by mkaruza** [comment]
I hereby contribute this patch under the BSD-new license

File: scripts/wsrep_sst_mariabackup.sh [hunk 1]
```diff
@@ -644,6 +644,27 @@ send_donor()
...
 }
 
+monitor_process()
+{
+    local sst_stream_pid=$1
+
+    while true ; do
+
+        if ! ps --pid "${WSREP_SST_OPT_PARENT}" &>/dev/null; then
+            wsrep_log_error "Parent mysqld process (PID:${WSREP_SST_OPT_PARENT}) terminated unexpectedly."
```
[the remark this answers is not in the document]
Comment (mkaruza) [reply scripts/wsrep_sst_mariabackup.sh]: Not with dash.

[the remark this answers is not in the document]
Comment (mkaruza) [reply scripts/wsrep_sst_mariabackup.sh]: Not with dash. But beginning of script shebang is set to use bash.

**Pull request merged**